components/x-topic-search/src/TopicSearch.jsx

```diff
@@ -1,95 +1,125 @@
-	async checkInput(event) {
-		const searchTerm = event.target.value && event.target.value.trim();
+		this.setState({ searchTerm });
 
-		if (searchTerm.length >= minSearchLength) {
-			return debounceGetSuggestions(searchTerm, maxSuggestions, apiUrl, followedTopicIds)
-				.then(result => {
-					resultExists = true;
-					return { showResult: true, result, searchTerm };
+		if (searchTerm.length >= this.minSearchLength) {
+			this.getSuggestions(searchTerm, this.maxSuggestions, this.apiUrl, this.props.followedTopicIds)
+				.then(({ resultsForTerm, followedSuggestions, unfollowedSuggestions }) => {
+					this.setState({
+						resultsForTerm,
+						followedSuggestions,
+						unfollowedSuggestions,
+						showResult: true
+					});
 				})
 				.catch(() => {
-					resultExists = false;
-					return { showResult: false };
+					this.setState({
+						resultsForTerm: null,
+						showResult: false
+					});
 				});
 		} else {
-			resultExists = false;
-			return Promise.resolve({ showResult: false });
+			this.setState({
+				resultsForTerm: null,
+				showResult: false
+			});
 		}
-	},
+	}
 
-	topicFollowed (subjectId) {
-		if (!followedTopicIds.includes(subjectId)) {
-			followedTopicIds.push(subjectId);
+	handleInteractionOutside(event) {
+		if (!this.rootEl.contains(event.target)) {
+			this.setState({
+				showResult: false
+			});
 		}
+	}
 
-		return { followedTopicIds };
-	},
-
-	topicUnfollowed (subjectId) {
-		const targetIdIndex = followedTopicIds.indexOf(subjectId);
-
-		if (targetIdIndex > -1) {
-			followedTopicIds.splice(targetIdIndex, 1);
+	handleInputClickOrFocus() {
+		if (this.state.searchTerm.length >= this.minSearchLength) {
+			this.setState({
+				showResult: true
+			});
 		}
+	}
 
-		return { followedTopicIds };
-	},
-
-	selectInput (event) {
-		event.target.select();
-		return { showResult: resultExists };
-	},
-
-	hideResult() {
-		return { showResult: false };
+	render() {
+		const { followedTopicIds } = this.props;
+		const { csrfToken, followedSuggestions, resultsForTerm, searchTerm, showResult, unfollowedSuggestions } = this.state;
+
+		return (
+			<div className={ classNames(styles['container']) } ref={el => this.rootEl = el}>
+				<h2 className="o-normalise-visually-hidden">
+					Search for topics, authors, companies, or other areas of interest
+				</h2>
+
+				<label className="o-normalise-visually-hidden" htmlFor="topic-search-input">Search and add topics</label>
+				<div className={ classNames(styles["input-wrapper"]) }>
+					<i className={ classNames(styles["search-icon"]) }/>
+					<input
+						type="search"
+						id="topic-search-input"
+						placeholder="Search and add topics"
+						className={ classNames(styles["input"]) }
+						data-trackable="topic-search"
+						autoComplete="off"
+						onInput={this.handleInputChange}
+						onClick={this.handleInputClickOrFocus}
+						onFocus={this.handleInputClickOrFocus}
```

Comment:
Space consistency for these three brackets? Although I know you prefer without spaces... I love spaces...
```
onInput={ this.handleInputChange }
onClick={ this.handleInputClickOrFocus }
onFocus={ this.handleInputClickOrFocus }
```

Reply:
I don't feel strongly either way, but I tend to follow airbnb react lint rules, as (I believe) it's fairly widely adopted:

http://airbnb.io/javascript/react/#spacing

I agree it should be consistent though, but without an agreed standard (for all x-dash devs), this could go back and forward endlessly. We could raise a PR which would enable the lint rule for this and then let people argue in the PR until someone clicks merge 😈